pytket/binders/passes.cpp

```diff
@@ -265,23 +265,29 @@ PYBIND11_MODULE(passes, m) {
       .def(
           "to_dict",
           [](const BasePass &base_pass) {
-            return py::object(base_pass.get_config()).cast<py::dict>();
+            return py::cast(serialise(base_pass));
           },
           ":return: A JSON serializable dictionary representation of the Pass.")
       .def_static(
           "from_dict",
-          [](const py::dict &base_pass_dict) {
-            return json(base_pass_dict).get<PassPtr>();
+          [](const py::dict &base_pass_dict,
+
+             std::map<std::string, std::function<Circuit(const Circuit &)>>
+                 &custom_deserialisation) {
+            return deserialise(base_pass_dict, custom_deserialisation);
           },
           "Construct a new Pass instance from a JSON serializable dictionary "
-          "representation.")
+          "representation.",
```

Comment:
This needs more documentation of the `custom_deserialization` parameter.

Reply:
added